rclcpp/include/rclcpp/executor.hpp

```diff
@@ -362,7 +362,6 @@ class Executor
   /// The context associated with this executor.
   std::shared_ptr<rclcpp::Context> context_;
 
-private:
```

Comment:
Is this line being removed to allow access to something in particular? Instead, could there be mutators added to this (the base) instead?

I don't think it's wise to make that list public.

Reply:
I have made weak_nodes_ and guard_conditions_ lists protected to be able to access them in StaticSingleThreadedExecutor class.